test/unit/cpu/test-forallN.cxx

```diff
@@ -0,0 +1,402 @@
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJK, IMoment, IGroup, IZone>>,
+        RAJA::NestedPolicy<RAJA::ExecList<RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec>>>,
+    std::tuple<
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_JI, IMoment, IDirection>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IDirection, IGroup, IZone>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJK, IMoment, IGroup, IZone>>,
+        RAJA::NestedPolicy<
+            RAJA::ExecList<RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec>,
+            RAJA::Permute<RAJA::PERM_LKJI>>>,
+    std::tuple<
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJ, IMoment, IDirection>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJK, IDirection, IGroup, IZone>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IMoment, IGroup, IZone>>,
+        RAJA::NestedPolicy<
+            RAJA::ExecList<RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec, RAJA::seq_exec>,
+            RAJA::Tile<
+                RAJA::TileList<RAJA::tile_none, RAJA::tile_none,
+                               RAJA::tile_fixed<64>, RAJA::tile_fixed<64>>,
+                RAJA::Permute<RAJA::PERM_JKIL>>>>>;
+                
+INSTANTIATE_TYPED_TEST_CASE_P(Sequential, ForallNKernelTest, LTimesSequentialTypes); 
+
+#ifdef RAJA_ENABLE_OPENMP
+using LTimesOpenMPTypes = ::testing::Types<
+    std::tuple<
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJ, IMoment, IDirection>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IDirection, IGroup, IZone>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IMoment, IGroup, IZone>>,
+        RAJA::NestedPolicy<
+            RAJA::ExecList<
+                RAJA::seq_exec, RAJA::seq_exec, 
+                RAJA::seq_exec, RAJA::omp_for_nowait_exec>,
+            RAJA::OMP_Parallel<RAJA::Permute<RAJA::PERM_LKIJ>>>>,
+    std::tuple<
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_IJ, IMoment, IDirection>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IDirection, IGroup, IZone>>,
+        RAJA::View<double, RAJA::Layout<int, RAJA::PERM_KJI, IMoment, IGroup, IZone>>,
+        RAJA::NestedPolicy<
+            RAJA::ExecList<
+                RAJA::seq_exec, RAJA::seq_exec, 
+                RAJA::omp_collapse_nowait_exec, RAJA::omp_collapse_nowait_exec>,
+            RAJA::OMP_Parallel<
+                RAJA::Tile<
+                    RAJA::TileList<RAJA::tile_none, RAJA::tile_none,
+                                   RAJA::tile_none, RAJA::tile_fixed<16>>,
+                    RAJA::Permute<
+                        RAJA::PERM_LKIJ, 
+                        RAJA::Execute // implict
+                        >>>>>>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(OpenMP, ForallNKernelTest, LTimesOpenMPTypes); 
+#endif
+
+#if defined(RAJA_ENABLE_AGENCY)
+// TODO: Write code here, depends on permute and tiling impl
```

Comment:
I haven't written tests here yet because I'm not totally sure what a meaningful translation of the OpenMP version would be. We don't have an equivalent for omp_collapse_nowait_exec with Agency. I might end up just trying a direct translation (or as close as we can get) unless there are any suggestions.